webapi/Skills/ChatSkills/ChatSkill.cs

```diff
@@ -383,9 +383,10 @@ private async Task<ChatMessage> GetChatResponseAsync(string chatId, string userI
         var documentMemories = tasks[1];
 
         // Fill in the chat history if there is any token budget left
-        var chatContextComponents = new List<string>() { chatMemories, documentMemories, planResult };
+        var chatContextComponents = new List<string>() { chatMemories, documentMemories };
         var chatContextText = string.Join("\n\n", chatContextComponents.Where(c => !string.IsNullOrEmpty(c)));
-        var chatHistoryTokenLimit = remainingToken - TokenUtilities.TokenCount(chatContextText);
+        var chatHistoryTokenLimit = remainingToken - TokenUtilities.TokenCount(chatContextText) - TokenUtilities.TokenCount(planResult);
```

Comment:
`ExtractChatHistoryAsync` will use all of `chatHistoryTokenLimit`, leaving no token for the plan.

Reply:
`TokenUtilities.TokenCount(planResult)` accounts for the remaining portion at the end.

Reply:
Saw that the number of tokens the plan used was taken out. Resolving this comment.